compiler/src/dmd/errors.d

```diff
@@ -457,8 +584,17 @@ extern (C++) void verrorReport(const ref Loc loc, const(char)* format, va_list a
         {
             info.headerColor = Classification.error;
             verrorPrint(format, ap, info);
+
+            // Hook: Generate SARIF report only if the --sarif flag is set
+            if (global.params.sarif)
+            {
+                generateSarifReport(info.loc, format, ap, info.kind);
+            }
             if (global.params.v.errorLimit && global.errors >= global.params.v.errorLimit)
+            {
+                fprintf(stderr, "error limit (%d) reached, use `-verrors=0` to show all\n", global.params.v.errorLimit);
```

Comment:
why is this change made?

Reply:
This was added to prevent the compiler from flooding the user with too many errors in case of multiple issues in the same file. For example, if a file has many undefined variables or other repetitive errors, the compiler will stop reporting after a set limit (e.g., 5 errors) and prompt the user to increase the limit with the `-verrors=0` option if they wish to see all errors.

If this feature is not needed, I can remove it to ensure that all errors are displayed without any restriction (like before)

Reply:
> If this feature is not needed, I can remove it to ensure that all errors are displayed without any restriction (like before)

Given SARIF is intended to be machine read, I think this is a good idea.